### PR TITLE
Adds flag zipkin.localServiceName to override any annotations

### LINKED ISSUE #67: Add feature to fix bad endpoints
Finagle apps routinely set the service name incorrectly for clients. Usually, they set the service name the remote service name. This causes data problems everywhere, not limited to the dependency graph. It is likely a good idea to provide an override flag that people can use at startup to force a local service name for all data.

See https://groups.google.com/forum/#!topic/zipkin-user/EgTN97wO4Sk

### PR DESCRIPTION
**Pull request by adriancole**
This should be used in general and will result in more consistent
trace relationships including the service dependency graph.

Fixes #67

### CODE REVIEW

File: core/src/test/java/zipkin2/finagle/ZipkinTracerTest.java [hunk 1]
```diff
@@ -53,21 +53,58 @@
...
+        return 1.0f;
+      }
+    };
+    tracer = newTracer(FakeSender.create().onSpans(spansSent::add));
+
+    recordClientSpan(root);
+
+    assertThat(spansSent.take().stream())
+        .flatExtracting(Span::localServiceName)
+        .containsExactly("favstar");
```
Comment (devinsba) [inline core/src/test/java/zipkin2/finagle/ZipkinTracerTest.java]: Is it appropriate here to set `remoteServiceName` to the value we are overwriting, in this case `web`?

Comment (adriancole) [reply core/src/test/java/zipkin2/finagle/ZipkinTracerTest.java]: It is a heuristic we could use.. In the case that the serviceName we are overriding was actually just a bad local service name it wouldn't help. OTOH zipkin treats the server's service name authoritative..

Comment (adriancole) [reply core/src/test/java/zipkin2/finagle/ZipkinTracerTest.java]: looped the idea back into the zipkin-user thread

**Comment by adriancole** [comment]
I'll just implement it like the way said as I don't feel like tracking this issue

**Comment by adriancole** [comment]
OK I've updated this.. if someone wants to try this branch you can setup maven like this

https://jitpack.io/#openzipkin/zipkin-finagle/localServiceName-SNAPSHOT

or something like this:

```bash
curl -sSL https://jitpack.io/com/github/openzipkin/zipkin-finagle/zipkin-finagle-http_2.12/${TAG}/zipkin-finagle-http_2.12-${TAG}.jar > zipkin-finagle-http_2.12.jar
```

File: core/src/main/java/zipkin2/finagle/MutableSpan.java [hunk 2]
```diff
@@ -111,22 +111,29 @@ synchronized MutableSpan setAddress(String key, InetSocketAddress ia) {
...
     for (V1Annotation a : annotations) {
       Endpoint ep = Endpoints.boundEndpoint(a.endpoint());
       span.addAnnotation(a.timestamp(), a.value(), ep.toBuilder().serviceName(service).build());
     }
     for (V1BinaryAnnotation ann : binaryAnnotations) {
-      if (ann.stringValue() == null) continue;
       Endpoint ep = Endpoints.boundEndpoint(ann.endpoint());
-      span.addBinaryAnnotation(
-          ann.key(), ann.stringValue(), ep.toBuilder().serviceName(service).build());
+      if (ann.stringValue() == null) { // service annotation
```
Comment (adriancole) [inline core/src/main/java/zipkin2/finagle/MutableSpan.java]: This was a bug. We had no unit tests reading back the address details. Now we do

**Comment by tmichnik1981** [comment]
We tested those changes and it looks good to us.

**Comment by adriancole** [comment]
@crispywalrus any commentary on this one?

**Comment by crispywalrus** [comment]
this looks good to me. are you going to add any t sets?

**Comment by adriancole** [comment]
yep tests pending. just wanted to make sure they would be worthwhile ;)

thanks for having a look

**Comment by adriancole** [comment]
added integration test, but there's something weird about the http tests.. I think there is some unrelated state problem. will look into it later.

**Comment by adriancole** [comment]
the state issue was on my laptop. all good

**Pull request merged**